tests/std/tests/GH_004388_unordered_meow_operator_equal/test.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <map>
+#include <set>
+#include <unordered_map>
+#include <unordered_set>
+
+using namespace std;
+
+namespace modulo {
+    // Functors for comparison and hashing
+    // based on modular arithmetic equivalence classes
+    template <int N>
+    struct compare {
+        bool operator()(int lhs, int rhs) const {
+            return (lhs % N) < (rhs % N);
+        }
+    };
+
+    template <int N>
+    struct equals {
+        bool operator()(int lhs, int rhs) const {
+            return (lhs % N) == (rhs % N);
+        }
+    };
+
+    template <int N>
+    struct hash {

```

Comment:
This is really, really nitpicking, but even though this is not technically shadowing due to the `modulo` namespace, having the lowercase `hash` floating around makes me a bit nervous. I think it's best for test code to be visually distinct from product code, and I don't see the namespace as serving an important purpose here. These types would be clearer as `ModLess`, `ModEqual`, and `ModHash`.